AWSCore/Service/AWSService.m

```diff
@@ -426,7 +426,10 @@ - (void) setRegion:(AWSRegionType)regionType service:(AWSServiceType)serviceType
     _regionType = regionType;
     _serviceType = serviceType;
     _regionName = [AWSEndpoint regionNameFromType:regionType];
-    _serviceName = [self serviceNameFromType:serviceType];
```

Comment:
is there a reason it was being overwritten here?

Reply:
`serviceName` is used in several places like [signing requests](https://github.com/aws-amplify/aws-sdk-ios/blob/24ecaabd4f182f44bce00eef63ca600c1ea4aac1/AWSCore/Authentication/AWSSignature.m#L203), [creating the host url](https://github.com/aws-amplify/aws-sdk-ios/blob/24ecaabd4f182f44bce00eef63ca600c1ea4aac1/AWSCore/Service/AWSService.m#L655) etc. I believe the `serviceName` is expected to be a constant for a service. Hence, we are always setting it to the same value defined  [here](https://github.com/aws-amplify/aws-sdk-ios/blob/24ecaabd4f182f44bce00eef63ca600c1ea4aac1/AWSCore/Service/AWSService.m#L262)

Reply:
The method signature is `setRegion:(AWSRegionType)regionType service:(AWSServiceType)serviceType` ie, it is changing the region and service type. Since serviceType is changed, the service name should also be updated right? ie the value written in initializer should be overriden.